contracts/extensions/MultisigPermissions.sol

```diff
@@ -0,0 +1,673 @@
+      return;
+    }
+
+    motion.executed = true;
+    bool overallSuccess = true;
+
+    for (uint256 i = 0; i < motion.data.length; i += 1) {
+      // solhint-disable-next-line avoid-low-level-calls
+      (bool success, ) = address(motion.targets[i]).call(motion.data[i]);
+      overallSuccess = overallSuccess && success;
+
+      // Allow failing execution after seven days
+      require(
+        success || motion.overallApprovalTimestamp + 7 days <= block.timestamp,
+        "colony-multisig-failed-not-one-week"
+      );
+    }
+
+    emit MotionExecuted(msgSender(), _motionId, overallSuccess);
+  }
+
+  function checkThreshold(uint256 _motionId, Vote _vote) private view returns (bool thresholdMet) {
+    Motion storage motion = motions[_motionId];
+    uint8 roleIndex;
+    // While there are still relevant roles we've not checked yet
+    while (uint256(motion.requiredPermissions) >= (1 << roleIndex)) {
+      // For the current role, is it required for the motion?
+      if ((motion.requiredPermissions & bytes32(1 << roleIndex)) > 0) {
+        uint256 threshold = getDomainSkillRoleThreshold(
+          motion.domainSkillId,
+          ColonyRole(roleIndex)
+        );
+        if (motionVoteCount[_motionId][_vote][ColonyRole(roleIndex)] < threshold) {
+          return false;
+        }
+      }
+      roleIndex += 1;
+    }
+    return true;
+  }
+
+  function getGlobalThreshold() public view returns (uint256) {
+    return globalThreshold;
+  }
+
+  function getMotionCount() public view returns (uint256) {
+    return motionCount;
+  }
+
+  function getMotion(uint256 motionId) public view returns (Motion memory) {
+    return motions[motionId];
+  }
+
+  function getMotionRoleVoteCount(
+    uint256 _motionId,
+    ColonyRole _role,
+    Vote _vote
+  ) public view returns (uint256) {
+    return motionVoteCount[_motionId][_vote][_role];
+  }
+
+  function getUserVote(
+    uint256 _motionId,
+    address _user,
+    ColonyRole _permission,
+    Vote _vote
+  ) public view returns (bool) {
+    bytes32 userVotes = motionVotes[_motionId][_user][_vote];
+    return (userVotes >> uint8(_permission)) & bytes32(uint256(1)) == bytes32(uint256(1));
+  }
+
+  function getDomainSkillRoleCounts(
+    uint256 _domainSkillId,
+    uint8 role
+  ) public view returns (uint256) {
+    return domainSkillRoleCounts[_domainSkillId][role];
+  }
+
+  function getDomainSkillThreshold(uint256 _domainSkillId) public view returns (uint256) {
+    return domainSkillThreshold[_domainSkillId];
+  }
+
+  function getDomainSkillRoleThreshold(
+    uint256 _domainSkillId,
+    ColonyRole _role
+  ) public view returns (uint256) {
+    if (domainSkillThreshold[_domainSkillId] > 0) {
+      return domainSkillThreshold[_domainSkillId];
+    }
+
+    if (globalThreshold > 0) {
+      return globalThreshold;
+    }
+
+    return (domainSkillRoleCounts[_domainSkillId][uint8(_role)] / 2) + 1;
+  }
+
+  function getActionSummary(
```

Comment:
Can we adapt the implementation to use the shared `GetActionSummary` mixin? Seems wasteful to have a slightly different reimplementation here.